applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -280,22 +292,27 @@ static void stream_enabled_cb(struct bt_audio_stream *stream)
 	if (ep_state_check(headsets[channel_index].sink_stream.ep, BT_AUDIO_EP_STATE_ENABLING) &&
 	    ep_state_check(headsets[channel_index].source_stream.ep, BT_AUDIO_EP_STATE_ENABLING)) {
 #endif /* CONFIG_STREAM_BIDIRECTIONAL */
-		ret = bt_audio_stream_start(&headsets[channel_index].sink_stream);
+		work_data.channel_index = channel_index;
+		work_data.retries = 0;
+		work_data.dir = BT_AUDIO_DIR_SINK;
+
+		ret = k_msgq_put(&kwork_msgq, &work_data, K_NO_WAIT);
 		if (ret) {
-			LOG_ERR("Failed to start %s headset sink stream",
-				headsets[channel_index].ch_name);
-			return;
+			LOG_ERR("No space in the queue for the bond");
 		}
-
+		k_work_schedule(&headsets[channel_index].stream_start_work, K_MSEC(20));
 #if CONFIG_STREAM_BIDIRECTIONAL
-		ret = bt_audio_stream_start(&headsets[channel_index].source_stream);
+		work_data.channel_index = channel_index;
+		work_data.retries = 0;
+		work_data.dir = BT_AUDIO_DIR_SOURCE;
+
+		k_work_flush_delayable(&headsets[channel_index].stream_start_work,
```

Comment:
I may see once I get further down, but it is not clear to me why the flush is located here.